public/app/features/explore/Logs/Logs.tsx

```diff
@@ -640,353 +622,299 @@ class UnthemedLogs extends PureComponent<Props, State> {
     return { from: firstTimeStamp, to: lastTimeStamp };
   });
 
-  scrollToTopLogs = () => {
+  const scrollToTopLogs = () => {
     if (config.featureToggles.logsInfiniteScrolling) {
-      if (this.state.logsContainer) {
-        this.state.logsContainer.scroll({
+      if (logsContainer) {
+        logsContainer.scroll({
           behavior: 'auto',
           top: 0,
         });
       }
     }
-    this.topLogsRef.current?.scrollIntoView();
+    topLogsRef.current?.scrollIntoView();
```

Comment:
Aprox from line 314 to 634, these used to be class methods that would not be recreated on every render. Could you please wrap them around `useCallback()`. The reason is to maintain the previous behavior and to not cause performance regressions and/or unnecessary re-renders.

Reply:
Made memoized utils static in 5d2f0ebd841c3343665712e73099f4a6e38f3f37 and added wrappers in fed463c46154ba0ac7442c745b21066e9b2b4d95